packages/qwik/src/core/render/ssr/render-ssr.ts

```diff
@@ -915,10 +926,20 @@ export const escapeAttr = (s: string) => {
   });
 };
 
-export const listenersNeedId = (listeners: Listener[]) => {
+const listenersNeedId = (listeners: Listener[]) => {
   return listeners.some((l) => l[1].$captureRef$ && l[1].$captureRef$.length > 0);
 };
 
+const addDynamicSlot = (hostCtx: QContext, elCtx: QContext) => {
+  let dynamicSlots = hostCtx.$dynamicSlots$;
+  if (!dynamicSlots) {
+    hostCtx.$dynamicSlots$ = dynamicSlots = [];
+  }
+  if (!dynamicSlots.includes(elCtx)) {
```

Comment:
O(n), is there a case that this can cause performance issues?

Reply:
N should be small! HashMaps get faster as N get bigger, for small N, array usually performs better. 
Also hashmap better if in-order removing is important.